### ISSUE TITLE
Unnecessary copy of edgelist?

### ISSUE THREAD

**Issue by Stein-ErikG** [ShobiStassen/PARC#43]
Hey,

I've been looking at the PARC code, and i noticed that in lines 298-306 (PARC/parc/_parc.py), you first create edgelist and then make a copy of edgelist. The copy is then used in the following code (see below). I would argue that this would put unnecessary strain on the memory? I have no idea how big these lists are an how much memory these would take up, tough. 

edgelist = list(zip(sources, targets))
edgelist_copy = edgelist.copy()
G = ig.Graph(edgelist, edge_attrs={'weight': csr_array.data.tolist()})
sim_list = G.similarity_jaccard(pairs=edgelist_copy)

Ive tested a version of the code, just using 'list(zip(sources, targets))' in place of edgelist and edgelist_copy, and this works fine. Maybe this could make the code a bit more memory efficient? 

G = ig.Graph(list(zip(sources, targets)), edge_attrs={'weight': csr_array.data.tolist()})
sim_list = G.similarity_jaccard(pairs=list(zip(sources, targets)))

Best,
Stein-Erik

**Comment by ahill187** [comment]
Yes I noticed that too, it seems unnecessary but I wasn't sure? 
It definitely takes up a lot of memory when `knn` is large and the number of samples is large.
